java/gandiva/src/test/java/org/apache/arrow/gandiva/evaluator/ProjectorTest.java

```diff
@@ -2599,4 +2599,33 @@ public void testInitCap() throws Exception {
     releaseRecordBatch(batch);
     releaseValueVectors(output);
   }
+
+  public class DummySecondaryCache implements JavaSecondaryCacheInterface {
+    public BufferResult getSecondaryCache(long addrKey, long sizeKey) {
+      return null;
+    }
+
+    public void setSecondaryCache(long addrKey, long sizeKey, long addrValue, long sizeValue) {
+    }
+  }
+
+  @Test
+  public void testSecondaryCache() throws GandivaException {
+    Field a = Field.nullable("a", int32);
+    Field b = Field.nullable("b", int32);
+    List<Field> args = Lists.newArrayList(a, b);
+
+    Field retType = Field.nullable("c", int32);
+    ExpressionTree root = TreeBuilder.makeExpression("add", args, retType);
+
+    List<ExpressionTree> exprs = Lists.newArrayList(root);
+
+    Schema schema = new Schema(args);
+
+    DummySecondaryCache secondaryCache = new DummySecondaryCache();
+
+    Projector eval = Projector.make(schema, exprs,
```

Comment:
Verify that the DummySecondaryCache Get And Set methods are called by setting boolean flags.
Also add the evaluation and result checking for completeness sake